### PR TITLE
Running all 3rd party integration tests when a submodule is changed

### PR DESCRIPTION
**Pull request by rafaelwestphal**
## Description
Integration tests should not be skipped when a submodule is upgraded. Enforcing that any changes to submodule/* folder will trigger the integration tests.

## How has this been tested?
I added a submodule change and verified that the tests were scheduled. Then i reverted the change.

## Checklist:
- Unit tests
  - [x] Unit tests do not apply.
  - [ ] Unit tests have been added/modified and passed for this PR.
- Integration tests
  - [x] Integration tests do not apply.
  - [ ] Integration tests have been added/modified and passed for this PR.
- Documentation
  - [x] This PR introduces no user visible changes.
  - [ ] This PR introduces user visible changes and the corresponding documentation change has been made.
- Minor version bump
  - [x] This PR introduces no new features.
  - [ ] This PR introduces new features, and there is a separate PR to bump the [minor version](https://github.com/GoogleCloudPlatform/ops-agent/blob/master/VERSION) since the last [release](https://github.com/GoogleCloudPlatform/ops-agent/releases) already.
  - [ ] This PR bumps the version.

### CODE REVIEW

**Comment by jefferbrecht** [comment]
LGTM 👍 

WDYT about also re-running the 3p tests if `third_party_apps_test.go` is modified?

**Comment by rafaelwestphal** [comment]
> LGTM 👍
> 
> WDYT about also re-running the 3p tests if `third_party_apps_test.go` is modified?
I thought about this, but didn't come up with a critical scenario where this would be an issue. I think if third_party_apps_test.go gets modified it doesn't necessarily needs all tests to be run? That being said, i don't usually touch third party apps.

File: integration_test/third_party_apps_test.go [hunk 1]
```diff
@@ -658,6 +658,17 @@ func modifiedFiles(t *testing.T) []string {
 // Checks the extracted app names against the set of all known apps.
 func determineImpactedApps(mf []string, allApps map[string]metadata.IntegrationMetadata) map[string]bool {
 	impactedApps := make(map[string]bool)
+	for _, f := range mf {
+		// File names: submodules/fluent-bit
+		if strings.HasPrefix(f, "submodules/") {
+			for app, _ := range allApps {
+				impactedApps[app] = true
+			}
+			log.Printf("impacted apps: %v", impactedApps)
```
Comment (martijnvans) [inline integration_test/third_party_apps_test.go:667]: I would do this as:
```
defer log.Printf("impacted apps: %v", impactedApps)
```
right after declaring impactedApps. Then you don't have to repeat this here and down below.

Comment (rafaelwestphal) [reply integration_test/third_party_apps_test.go]: Done!

**Review by martijnvans — APPROVED** [review]

**Review by jefferbrecht — APPROVED** [review]

**Comment by martijnvans** [comment]
test failures are unrelated, and the still-running windows tests passed on a previous snapshot. merging.

**Pull request merged**